### ISSUE TITLE
Things that I needed to fix to get the inference working:

### ISSUE THREAD

**Issue by SpirosK** [sail-sg/sewformer#21]
--- a/Sewformer/models/transformer.py
+++ b/Sewformer/models/transformer.py
@@ -227,7 +227,7 @@ class TransformerDecoderLayer(nn.Module):
                                    key_padding_mask=memory_key_padding_mask)[0]
         tgt = tgt + self.dropout2(tgt2)
         tgt = self.norm2(tgt)
-        tgt2 = self.linear2(slef.dropout(self.activation(self.linear1(tgt))))
+        tgt2 = self.linear2(self.dropout(self.activation(self.linear1(tgt))))

--- a/Sewformer/experiment.py
+++ b/Sewformer/experiment.py
@@ -246,7 +246,7 @@ class ExperimentWrappper(object):

         # Dataset
         data_class = getattr(data, data_config['class'])
-        dataset = data_class(data_root, data_config, gt_caching=True, feature_caching=False)
+        dataset = data_class(root_dir=data_root, sim_root=data_root, start_config=data_config, gt_caching=True, feature_caching=False)

--- a/Sewformer/requirements.txt
+++ b/Sewformer/requirements.txt
@@ -11,4 +11,5 @@ entmax
 numba
 pytz
 cairosvg
-
+wmi
+scikit-learn

--- a/Sewformer/configs/test.yaml
+++ b/Sewformer/configs/test.yaml
@@ -8,7 +8,8 @@ experiment:
   run_name: Detr2d-V6-final-dif-ce-focal-schd-agp
   run_id:
   local_dir: ./garment_outputs/Detr2d-V6-final-dif-ce-focal-schd-agp
-
+  is_training: False
+

You are welcome :)

**Comment by SpirosK** [comment]
--- a/Sewformer/models/transformer.py
+++ b/Sewformer/models/transformer.py
@@ -227,7 +227,7 @@ class TransformerDecoderLayer(nn.Module):
                                    key_padding_mask=memory_key_padding_mask)[0]
         tgt = tgt + self.dropout2(tgt2)
         tgt = self.norm2(tgt)
-        tgt2 = self.linear2(slef.dropout(self.activation(self.linear1(tgt))))
+        tgt2 = self.linear2(self.dropout(self.activation(self.linear1(tgt))))

--- a/Sewformer/experiment.py
+++ b/Sewformer/experiment.py
@@ -246,7 +246,7 @@ class ExperimentWrappper(object):

         # Dataset
         data_class = getattr(data, data_config['class'])
-        dataset = data_class(data_root, data_config, gt_caching=True, feature_caching=False)
+        dataset = data_class(root_dir=data_root, sim_root=data_root, start_config=data_config, gt_caching=True, feature_caching=False)


--- a/Sewformer/requirements.txt
+++ b/Sewformer/requirements.txt
@@ -11,4 +11,5 @@ entmax
 numba
 pytz
 cairosvg
-
+wmi
+scikit-learn

--- a/Sewformer/configs/test.yaml
+++ b/Sewformer/configs/test.yaml
@@ -8,7 +8,8 @@ experiment:
   run_name: Detr2d-V6-final-dif-ce-focal-schd-agp
   run_id:
   local_dir: ./garment_outputs/Detr2d-V6-final-dif-ce-focal-schd-agp
-
+  is_training: False
+


You are welcome :)
